pyserini/index/_base.py

```diff
@@ -260,6 +260,35 @@ def get_document_vector(self, docid: str) -> Optional[Dict[str, int]]:
             doc_vector_dict[term] = doc_vector_map.get(JString(term.encode('utf-8')))
         return doc_vector_dict
 
+    def get_term_positions(self, docid: str) -> Optional[Tuple[Dict[str, int], str]]:
+        """Return the term position mapping of the document with ``docid`` and the recovered document using the list. Note that
+        the term in the document is stemmed and stop words may be removed according to your index settings. Also,
+        requesting the document vector of a ``docid`` that does not exist in the index will return ``None`` (as opposed
+        to an empty dictionary); this forces the caller to handle ``None`` explicitly and guards against silent errors.
+
+        Parameters
+        ----------
+        docid : str
+            Collection ``docid``.
+
+        Returns
+        -------
+        Optional[Tuple[Dict[str, int], str]]
+            A tuple contains a dictionary with analyzed terms as keys and corresponding posting list as values, and a
+            string representing the recovered document
```

Comment:
Why do you want to return "string representing the recovered document"? What's the use case? Users can easily do this also if they want?

Reply:
Currently, the document returned by `doc.contents()` is the document before it is processed by Lucene. The reconstructed document will provide a view of the document after Lucene's processing. 
Yes. Users can easily do this if they want, and not every user needs this. I think we can move this piece of code to `usage-indexreader.md` as an example of how to use this function.

Reply:
Agreed - can you do that?

Reply:
done